### PR TITLE
V4

### PR DESCRIPTION
**Pull request by AlbertoElias**
Assigning this to myself as it's technically Kaelig's pr

### CODE REVIEW

File: bookmarklet/bookmarklet.js [hunk 1]
```diff
@@ -171,14 +165,21 @@
 		}
 	};
 
+	var head = document.getElementsByTagName("head")[0];
+	var gridScript = document.createElement("script");
+	gridScript.src = "http://build.origami.ft.com/bundles/js?modules=o-grid@^4.0.0-beta.6";
```
Comment (AlbertoElias) [inline bookmarklet/bookmarklet.js]: Change this to ^4.0.0

File: demos/src/config.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-'use strict';
```
Comment (AlbertoElias) [inline demos/src/config.js]: @expe This isn't an ES6 module, is ESLint treating it that way?

File: demos/src/js/almost-equal.js [hunk 3]
```diff
@@ -15,8 +14,3 @@ function almostEqual(a, b, absoluteError, relativeError) {
 	}
 	return a === b;
 }
-
-almostEqual.FLT_EPSILON = 1.19209290e-7;
```
Comment (AlbertoElias) [inline demos/src/js/almost-equal.js]: @expe Why did you remove this?

File: main.js [hunk 4]
```diff
@@ -1,6 +1,59 @@
...
+	if (isIE8) {
+		return '20px';
+	}
+
+	return getGridProperties().gutter;
+}
 
 module.exports = {
-	getCurrentLayout: require('./src/js/getCurrentLayout')
+	getCurrentLayout: getCurrentLayout,
```
Comment (AlbertoElias) [inline main.js]: @expe can be exported in an es6 module format

File: src/scss/_main.scss [hunk 5]
```diff
@@ -6,46 +6,71 @@
...
+/// @param {String} $layout-name - Name of the layout (e.g. S)
+/// @param {Number} $layout-width - Layout width in px
+/// @param {Number} $gutter-width [null] - Gutter width in px
+@mixin oGridAddLayout($layout-name, $layout-width, $gutter-width: null) {
 	$temp-layouts: ();
+	$temp-gutters: (default: #{oGridGutter()});
 
 	// Add the new layout in the correct position:
-	// (we want $o-grid-layouts to ordered from the smallest to the largest layout)
+	// (we want $o-grid-layouts and $o-grid-gutters to ordered from the smallest to the largest layout)
```
Comment (AlbertoElias) [inline src/scss/_main.scss]: be ordered

File: src/scss/_main.scss [hunk 6]
```diff
@@ -6,46 +6,71 @@
...
-	// When snappy mode is enabled, force $_o-grid-max-width to the largest layout width
-	// instead of the default $_o-grid-max-width
-	@if $o-grid-mode == 'snappy' {
-		$_o-grid-max-width: map-get($o-grid-layouts, nth($_o-grid-layout-names, -1)) !global;
+/// Get the gutter of a layout
+///
+/// @param {String|null|Boolean} $layout-name - One of $o-grid-layouts
+@function oGridGutter($layout-name: default) {
+	// Falsy values mean 'default'
+	@if not $layout-name {
```
Comment (AlbertoElias) [inline src/scss/_main.scss]: Isn't it already receiving a default value?

File: demos/src/config.js (hunk 2 above)
Comment (expe) [inline demos/src/config.js]: Yes, ESLint is complaining because we have in our default rules:
```
"ecmaFeatures": {
        "modules": true
    }
```
I believe Babel adds 'use strict' when transpiled, although it doesn't look like we're doing any transpiling in o-grid. Would this be handled by OBT or should I override the above rule?

File: demos/src/js/almost-equal.js (hunk 3 above)
Comment (expe) [inline demos/src/js/almost-equal.js]: It was removed by Kaelig, I don't know what this is.

**Comment by AlbertoElias** [comment]
@expe Oh, I was going to fix the other ones, but thanks!
